### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -8,7 +15,24 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
-    public String[] readFromFile(String fileName) {
-        return null;
+    public String[] readFromFile(String fileName) throws IOException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:18]: let's handle this exception with try catch

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -8,7 +15,24 @@
...
+
+        List<String> lines = Files.readAllLines(Paths.get(fileName));
+        List<String> listResult = new ArrayList<>();
+        for (String line : lines) {
+            String[] splitLine = line.split(" ");
+            for (String word : splitLine) {
+                if (word.charAt(0) == 'w' || word.charAt(0) == 'W') {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:25]: do we need to check both of this cases?

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/FileWork.java]: 1.added try - catch block.

File: src/main/java/core/basesyntax/FileWork.java (hunk 2 above)
Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/FileWork.java]: 2. simplified if condition.

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -9,6 +16,29 @@
...
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+
+        List<String> lines;
+        try {
+            lines = Files.readAllLines(Paths.get(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:25]: let's imagine i'm using this method and it's couldn't find a file and throw an exception, but program will continue to work and i will just get the null(not knowing that exception been thrown)

Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/FileWork.java]: done.
changed "catch" block

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -9,6 +16,25 @@
  */
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:19]: remove empty line

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -9,6 +16,25 @@
...
+            String[] splitLine = line.split(" ");
+            for (String word : splitLine) {
+                if (word.toLowerCase().charAt(0) == 'w') {
+                    listResult.add(word.replaceAll("\\W+", "").toLowerCase());
+                }
+            }
+        }
+        String[] result = listResult.toArray(String[]::new);
+        Arrays.sort(result);
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:37]: same

File: src/main/java/core/basesyntax/FileWork.java [hunk 6]
```diff
@@ -9,6 +16,25 @@
...
+        List<String> listResult = new ArrayList<>();
+        for (String line : lines) {
+            String[] splitLine = line.split(" ");
+            for (String word : splitLine) {
+                if (word.toLowerCase().charAt(0) == 'w') {
+                    listResult.add(word.replaceAll("\\W+", "").toLowerCase());
+                }
+            }
+        }
+        String[] result = listResult.toArray(String[]::new);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:35]: you can use Collections.sort()

**Pull request closed without merging**